include/url.h

```diff
@@ -75,24 +75,64 @@ namespace Url
             , fragment_(other.fragment_)
             , userinfo_(other.userinfo_) { }
 
-        /**************************
-         * Component-wise access. *
-         **************************/
+        /**************************************
+         * Component-wise access and setting. *
+         **************************************/
         const std::string& scheme() const { return scheme_; }
+        Url& scheme(const std::string& s)
```

Comment:
I wouldn't overload accessors with mutators. For mutators, it is common to use a `set` prefix.